employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/shiftroster/ShiftEditForm.java

```diff
@@ -50,94 +49,72 @@
 import static org.optaweb.employeerostering.gwtui.client.common.EventManager.Event.SHIFT_ROSTER_INVALIDATE;
 
 @Templated
-public class ShiftEditForm implements IsElement {
+public class ShiftEditForm extends AbstractFormPopup {
 
-    @Inject
-    @DataField("root")
-    private HTMLDivElement root;
-
-    @Inject
-    @DataField("popup-title")
-    @Named("span")
-    private HTMLElement popupTitle;
-
-    @Inject
-    @DataField("close-button")
-    private HTMLButtonElement closeButton;
-
-    @Inject
     @DataField("from")
     private LocalDateTimePicker from;
 
-    @Inject
     @DataField("to")
     private LocalDateTimePicker to;
 
-    @Inject
     @DataField("spot")
     private ListBox spotSelect; //FIXME: Don't use GWT widget
 
-    @Inject
     @DataField("employee")
     private ListBox employeeSelect; //FIXME: Don't use GWT widget
 
-    @Inject
     @DataField("pinned")
     private HTMLInputElement pinned;
 
-    @Inject
     @DataField("delete-button")
     private HTMLButtonElement deleteButton;
 
-    @Inject
-    @DataField("cancel-button")
-    private HTMLButtonElement cancelButton;
-
-    @Inject
     @DataField("apply-button")
     private HTMLButtonElement applyButton;
 
-    @Inject
     private TenantStore tenantStore;
 
-    @Inject
-    private PopupFactory popupFactory;
-
-    @Inject
     private EventManager eventManager;
 
-    @Inject
     private TranslationService translationService;
 
-    private FormPopup formPopup;
-
     private ShiftGridObject shiftGridObject;
 
+    @Inject
+    public ShiftEditForm(PopupFactory popupFactory, HTMLDivElement root, @Named("span") HTMLElement popupTitle, HTMLButtonElement closeButton,
+                         HTMLButtonElement cancelButton, LocalDateTimePicker from, LocalDateTimePicker to,
+                         ListBox spotSelect, ListBox employeeSelect, HTMLInputElement pinned,
+                         HTMLButtonElement deleteButton, HTMLButtonElement applyButton,
+                         TenantStore tenantStore, EventManager eventManager,
+                         TranslationService translationService) {
```

Comment:
Maybe one parameter per line?

Reply:
It does read better (and makes for easier refactoring). Done.